packages/documentation/copy/en/reference/JSX.md

```diff
@@ -70,14 +69,73 @@ This is important for two reasons:
 TypeScript uses the [same convention that React does](http://facebook.github.io/react/docs/jsx-in-depth.html#html-tags-vs.-react-components) for distinguishing between these.
 An intrinsic element always begins with a lowercase letter, and a value-based element always begins with an uppercase letter.
 
+### The `JSX` namespace
+
+JSX in TypeScript is typed by the `JSX` namespace. The `JSX` namespace may be defined in various places, depending on the `jsx` compiler option.
+
+The `jsx` options `preserve`, `react`, and `react-native` use the type definitions for classic runtime. This means a variable needs to be in scope that’s determined by the `jsxFactory` compiler option. The `JSX` namespace needs to specified on the top-most identifier of the JSX factory. For example, React uses the default factory `React.createElement`. This means its `JSX` namespace should be defined as `React.JSX`.
+
+```ts
+export function createElement(): any;
+
+export namespace JSX {
+  // …
+}
+```
+
+And the user should always import React as `React`.
+
+```ts
+import * as React from 'react';
+```
+
+Preact uses the JSX factory `h`. That means its types should be defined as the `h.JSX`.
+
+```ts
+export function h(props: any): any;
+
+export namespace h.JSX {
+  // …
+}
+```
+
+The user should use a named import to import `h`.
+
+```ts
+import { h } from 'preact';
+```
+
+For the `jsx` options `react-jsx` and `react-jsxdev`, the `JSX` namespace needs to be exported from the matching entry points. For `react-jsx` this is `${jsxImportSource}/jsx-runtime`. For `react-jsxdev`, this is `${jsxImportSource}/jsx-dev-runtime`. Since these don’t use a file extension, you must use the [`exports`](https://nodejs.org/api/packages.html#exports) field in `package.json` map in order to support ESM users.
+
+```json 
+{
+  "exports": {
+    "./jsx-runtime": "./jsx-runtime.js",
+    "./jsx-dev-runtime": "./jsx-dev-runtime.js",
+  }
+}
+```
+
+Then in `jsx-runtime.d.ts` and `jsx-dev-runtime.d.ts`:
+
+```ts
+export namespace JSX {
+  // …
+}
+```
+
+Note that while exporting the `JSX` namespace is sufficient for type checking, the production runtime needs the `jsx`, `jsxs`, and `Fragment` exports at runtime, and the development runtime needs `jsxDEV` and `Fragment`. Ideally you add types for those too.
+
```

Comment:
Since this is all specific to a single library, I don't think we need docs for that. The docs are equal to the current implementation in `@tyes/react` or `react`.

It was already problematic that these were introduced without working with React types maintainers. Writing docs make it seem like the TS team just need to update docs and React types maintainers just need to pay attention which is not (and [was not) healthy for the ecosystem](https://github.com/microsoft/TypeScript/pull/39199#issuecomment-731090590).

Reply:
`react-jsx` is specific to libraries other than React?

Reply:
None of the possible options are specific to React, but most have the word `react` in it.

| Name | Meaning |
| --- | --- |
| `preserve` | Preserve JSX, emit `.jsx` extension |
| `react-native` | Preserve JSX, emit `.js` extension |
| `react` | JSX classic runtime |
| `react-jsx` | JSX automatic runtime |
| `react-jsxdev` | JSX automatic development runtime |

Reply:
I understand that for classic because that's old. No idea why this was continued with `react-jsx`